src/runtime/String.h

```diff
@@ -21,6 +21,7 @@
 #define __EscargotString__
 
 #include "runtime/PointerValue.h"
+#include "api/EscargotPublic.h"
```

Comment:
`EscargotPublic.h` contains public APIs which are generally used in other systems or applications.
So, including this file inside the Escargot source seems unnatural.
How about defining another enum value in `String.h` which is similar to `WriteOptions` ?
`RegExpObjectRef::RegExpObjectOption` and `RegExpObject::Option` could be a reference for this case.
(I have fixed some minor bug in #571 )